### ISSUE TITLE
OpenMSX and OpenDingux (beta)

### ISSUE THREAD

**Issue by a-dekker** [openMSX/openMSX#1363]
About half a year ago I asked about OpenDingux releases in a now closed issue (https://github.com/openMSX/openMSX/issues/1302).

Maybe it is time to ask it once more: is a more recent openMSX release planned for OpenDingux? I am using OD beta now for quite some time, and it has become very mature. More and more users are switching to OD beta and the toolchain for OD beta is available as already concluded in https://github.com/openMSX/openMSX/issues/1302.

**Comment by MBilderbeek** [comment]
If there's a new official toolchain that I can install, which is supported by the community, I'd be happy to build 17.0 for OD again (and also resume development builds).

**Comment by a-dekker** [comment]
The latest toolchain can be found here: http://od.abstraction.se/opendingux/toolchain/opendingux-gcw0-toolchain.2021-03-10.tar.xz

**Comment by MBilderbeek** [comment]
@a-dekker  As far as I understand that's a nightly build (which can have any status)... Where to get the latest stable release?

EDIT: oh, it looks like it's from March. Stable after all? Will check it.

**Comment by a-dekker** [comment]
OD beta is developed in a kind of nightly build cycle. The device build is updated maybe every 14 days. The toolchain is updated less frequently (the release before March was Oct 2020 as I remember correctly). So afaik there is no official stable release (maybe once it loses it beta flag?) for now but a latest release (with stable quality ;-))

**Comment by MBilderbeek** [comment]
@a-dekker @mthuurne Looks like it's almost working... but when creating the OPK file this happens:
```
Creating OPK file:
mksquashfs \
	build/package-dingux/default.gcw0.desktop \
	share/icons/openMSX-logo-32.png \
	build/package-dingux/manual.man.txt \
	derived/mipsel-dingux-opt-3rd/bindist/* \
	derived/mipsel-dingux-opt-3rd/bindist/../openmsx-17.0.opk \
	-noappend -comp gzip -all-root
mksquashfs: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by mksquashfs)
mksquashfs: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /opt/gcw0-toolchain/bin/../lib/liblzma.so.5)
make[1]: *** [build/package-dingux/opk.mk:20: bindist] Error 1
make[1]: Leaving directory '/tmp/openmsx-17.0'
make: *** [build/main.mk:649: staticbindist] Error 2
sonata@20:21:/tmp/openmsx-17.0$ which mksquashfs

/opt/gcw0-toolchain/usr/bin/mksquashfs
```

Any ideas?

**Comment by MBilderbeek** [comment]
Aha:
```
$ /lib/x86_64-linux-gnu/libc.so.6 
GNU C Library (Debian GLIBC 2.31-12) stable release version 2.31.
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.
Compiled by GNU CC version 10.2.1 20210110.
libc ABIs: UNIQUE IFUNC ABSOLUTE
For bug reporting instructions, please see:
<http://www.debian.org/Bugs/>.
```
my libc is too old for that toolchain?

**Comment by a-dekker** [comment]
libc 2.31 does seem fairly recent (also the version I have on my rolling update Linux systems).

**Comment by a-dekker** [comment]
You could try the previous toolchain: https://ozgur.kazancci.com/jutleys/13-RS90/toolchain/opendingux-gcw0-toolchain.2020-10-01.tar.gz
I did successfully create some OPK files with this toolchain in the past. I have not used the March release yet.

**Comment by MBilderbeek** [comment]
Strange, I get the same error...

```
$ file /opt/gcw0-toolchain/bin/../lib/liblzma.so.5file: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /opt/gcw0-toolchain/bin/../lib/liblzma.so.5)
$ ls -l /opt/gcw0-toolchain/bin/../lib/liblzma.so.5
lrwxrwxrwx 1 root root 16 okt  1  2020 /opt/gcw0-toolchain/bin/../lib/liblzma.so.5 -> liblzma.so.5.2.5
 ls -l /opt/gcw0-toolchain/bin/../lib/liblzma.so.5.2.5 
-rwxr-xr-x 1 root root 178080 okt  1  2020 /opt/gcw0-toolchain/bin/../lib/liblzma.so.5.2.5
$ ls -l /opt/gcw0-toolchain_2021-03-10/bin/../lib/liblzma.so.5.2.5 
-rwxr-xr-x 1 manuel manuel 178080 mrt 10 22:43 /opt/gcw0-toolchain_2021-03-10/bin/../lib/liblzma.so.5.2.5
```

**Comment by a-dekker** [comment]
I do see the same in the Oct 2020 release. Perhaps I used the local mksquashfs and not the toolchain one for creating the OPK image.

**Comment by MBilderbeek** [comment]
OK, that works better (toolchain at the end of the PATH). Can you try attached OPK?
[openmsx-17.0.opk.gz](https://github.com/openMSX/openMSX/files/6728721/openmsx-17.0.opk.gz)

This is built with the October 2020 toolchain.

**Comment by a-dekker** [comment]
Thanks!

First impression: it does work, but with a bit of issues:
- Exiting OpenMSX freezes OpenMSX (hard kill needed)
- There seems to be some kind of blurring effect active? The picture is a bit less sharp than using 0.15.
- Performance feels a bit degraded. Playing Aleste 2 gives some stuttering scrolling. Does not happen in 0.15.
- Issues that seems to plague both 0.15 and 0.17 on OD beta: inserting roms or changing machine profile lead to a crash

**Comment by MBilderbeek** [comment]
OK, thanks for testing. So, I guess we shouldn't put it up for download then... Well, as I can't debug these issues, that's it for me. I do have the automatic build running again now, so new builds will be uploaded when they are made.

To fix these issues, we need someone able to debug them. Are you up for it? Or can you find a developer who is?

**Comment by a-dekker** [comment]
I could run stuff and provide (debug) logging, but debugging is not really my cup of tea. Someone like https://gitlab.com/gameblabla is the most active in porting emulators to handhelds at the moment, but I have no idea if he would be interested in OpenMSX.

**Comment by MBilderbeek** [comment]
@a-dekker did you ask him?

**Comment by a-dekker** [comment]
No, I didn't ask him in the end. If you ask me, I would say @mthuurne would have been the perfect person for this to dive in deeper back then, as he was/is involved in both OpenMSX and OpenDingux.
Nowadays OpenDingux (beta) is pretty much a dead platform and I see no point in keeping this issue open. So lets close this one as a "won't fix".
